SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,560 @@
+    this.updateField('requestType', event.target.value);
+  }
+
+  getScheduleType() {
+    if (this.isEditing() && !(this.props.form && this.props.form.scheduleType)) {
+      if (this.props.request.request[QUARTZ_SCHEDULE]) {
+        return QUARTZ_SCHEDULE;
+      } else {
+        return CRON_SCHEDULE;
+      }
+    } else {
+      if (this.props.form && this.props.form.scheduleType) {
+        return this.props.form.scheduleType;
+      } else {
+        return CRON_SCHEDULE;
+      }
+    }
+  }
+
+  renderRequestTypeSelectors() {
+    const tooltip = (
+      <ToolTip id="cannotChangeRequestTypeAfterCreation">Option cannot be altered after creation</ToolTip>
+    );
+    const selectors = REQUEST_TYPES.map((requestType, key) => {
+      const selector = (
+        <button
+          key={key}
+          value={requestType}
+          className={classNames('btn', 'btn-default', {active: this.getValue('requestType') === requestType})}
+          onClick={event => this.updateTypeButtonClick(event)}
+          disabled={this.getButtonsDisabled(requestType)}
+        >
+          {Utils.humanizeText(requestType)}
+        </button>
+      );
+      if (this.isEditing() && requestType === this.getValue('requestType')) {
+        return <OverlayTrigger placement="top" key={key} overlay={tooltip}>{selector}</OverlayTrigger>;
+      } else {
+        return selector;
+      }
+    })
+    return selectors;
+  }
+
+  renderLoadBalanced() {
+    const checkbox = (
+      <label htmlFor="load-balanced" className={classNames({subtle: this.isEditing()})}>
+        <CheckBox
+          id = "load-balanced"
+          onChange = {event => this.updateField("loadBalanced", !this.getValue("loadBalanced"))}
+          checked = {this.getValue("loadBalanced")}
+          disabled = {this.isEditing()}
+          noFormControlClass = {true}
+        />
+        {" Load balanced"}
+      </label>
+    );
+    let field;
+    if (this.isEditing()) {
+      field = (
+        <OverlayTrigger
+          placement="top"
+          overlay = {<ToolTip id="cannotChangeLoadBalancedAfterCreation">Option cannot be altered after creation</ToolTip>}>
+          {checkbox}
+        </OverlayTrigger>
+      );
+    } else {
+      field = checkbox;
+    }
+    return (
+      <div className = 'form-group'>
+        {field}
+      </div>
+    );
+  }
+
+  renderRequestTypeSpecificFormFields() {
+    return (
+      <div>
+        {
+          this.shouldRenderField('instances') ?
+            <div className='form-group'>
+              <label htmlFor='instances'>Instances</label>
+              <FormField
+                id = 'instances'
+                prop = {{
+                  updateFn: event => this.updateField('instances', event.target.value),
+                  placeholder: "1",
+                  inputType: 'text',
+                  value: this.getValue('instances')
+                }}
+              />
+            </div> :
+            undefined
```

Comment:
Should these `undefined`s be `null`s?
I know react renders both the same way, but I saw @kwm4385 using `null` for the same purpose.